extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/cdi/WithSpanInterceptor.java

```diff
@@ -53,35 +57,100 @@ public Object span(final ArcInvocationContext invocationContext) throws Exceptio
                 invocationContext.getParameters(),
                 invocationContext.getInterceptorBindings());
 
+        final Class<?> returnType = invocationContext.getMethod().getReturnType();
         Context parentContext = Context.current();
```

Comment:
@cescoffier On the WithSpanInterceptorTest, the parent context is not cleaned up after calling the 1st test, verifying the Uni. 
The next test, instead of seeing no parent, will see the 1st created span. 
Not sure if this is a problem of the test setup itself...

Reply:
Yeah... start in one thread and end in a different one. 
would be nice to start the span in the same thread as the Uni...

Reply:
There is no duplicated context in these tests. So, we are back to regular thread locals. 
But, yes, in the current state of the tests, I do not see where we switch to a different thread as the emission are synchronous.

Reply:
updated the above OpenTelemetryMpContextPropagationProvider to not restore the previous context only if there is an open span.